packages/extension/src/gui.view.ts

```diff
@@ -162,15 +165,49 @@ export class MarqueeGui extends EventEmitter {
          * the extension properly exports a setup method
          */
         extension.exports &&
-        extension.exports.marquee &&
-        typeof extension.exports.marquee.setup === 'function'
+        extension.exports.marquee
+      ) {
+        customWidgetCounter += extension.exports.marquee.customWidgetCounter || 1
+      }
+    }
+
+    for (const extension of widgets) {
+      /**
+       * continue if extension doesn't expose a marquee widget
+       * within its package.json
+       */
+      if (
+        !extension.packageJSON.marquee?.widget ||
+        /**
+         * don't show example widget extension in production
+         */
+        (process.env.NODE_ENV !== 'development' && extension.id === 'stateful.marquee')
       ) {
-        const defaultState = extension.exports.marquee?.disposable?.state || {}
-        const defaultConfiguration = extension.exports.marquee?.disposable?.configuration || {}
-        const ch = new Channel(extension.id, { ...defaultState, ...defaultConfiguration })
-        ch.registerPromise([this.panel.webview]).then((client) => {
-          extension.exports.marquee.setup(client)
-        })
+        continue
+      }
+
+      /**
+       * only setup a communication channel to the extension backend, if
+       */
+      if (
+      /**
+         * the extension is active so we can access its exported APIs
+         */
+        extension.isActive &&
+        /**
+         * the extension properly exports a setup method
+         */
+        extension.exports &&
+        extension.exports.marquee
+      ) {        
```

Comment:
same here, I would just put this into a single for loop and have some nested ifs

Reply:
I actually tried that at first, but didn't find a way to differentiate thirdPartywidgets to marquee since they all have a marquee object. A simpler solution can be to remove - do we need that last bit?

<img width="854" alt="Screen Shot 2022-08-25 at 10 28 58 AM" src="https://user-images.githubusercontent.com/63378463/186734837-e4336415-ba1f-4edd-8773-fad4b1bd47a5.png">

Reply:
nvm I found a better solution